Enhancements/TwilightPrincessHD_Bloom/rules.txt

```diff
@@ -0,0 +1,372 @@
+#name = 0% (Bloom will stay static)
+#condition = $bloom_enabled
+#$bloom_blur_strength = 0.0
+#
+#[Preset]
+#category = Periodic Bloom Blur
+#name = 10%
+#condition = $bloom_enabled
+#$bloom_blur_strength = 0.1
+#
+#[Preset]
+#category = Periodic Bloom Blur
+#name = 20%
+#condition = $bloom_enabled
+#$bloom_blur_strength = 0.2
+#
+#[Preset]
+#category = Periodic Bloom Blur
+#name = 30%
+#condition = $bloom_enabled
+#$bloom_blur_strength = 0.3
+#
+#[Preset]
+#category = Periodic Bloom Blur
+#name = 40%
+#condition = $bloom_enabled
+#$bloom_blur_strength = 0.4
+#
+#[Preset]
+#category = Periodic Bloom Blur
+#name = 50% (Recommended)
+#condition = $bloom_enabled
+#$bloom_blur_strength = 0.5
+#
+#[Preset]
+#category = Periodic Bloom Blur
+#name = 60%
+#condition = $bloom_enabled
+#$bloom_blur_strength = 0.6
+#
+#[Preset]
+#category = Periodic Bloom Blur
+#name = 70%
+#condition = $bloom_enabled
+#$bloom_blur_strength = 0.7
+#
+#[Preset]
+#category = Periodic Bloom Blur
+#name = 80%
+#condition = $bloom_enabled
+#$bloom_blur_strength = 0.8
+#
+#[Preset]
+#category = Periodic Bloom Blur
+#name = 90%
+#condition = $bloom_enabled
+#$bloom_blur_strength = 0.9
+#
+#[Preset]
+#category = Periodic Bloom Blur
+#name = 100% (Native)
+#condition = $bloom_enabled
+#default = 1
+#$bloom_blur_strength = 1.0
+
+
+# Area Specific Bloom Tint (49865bd2e62efda1)
+
+[Preset]
+category = Area Specific Bloom Tint
+name = 0% (Same white bloom color everywhere)
+condition = $bloom_enabled
+$bloom_tint_strength = 0.0
+
+[Preset]
+category = Area Specific Bloom Tint
+name = 10%
+condition = $bloom_enabled
+$bloom_tint_strength = 0.1
+
+[Preset]
+category = Area Specific Bloom Tint
+name = 20%
+condition = $bloom_enabled
+$bloom_tint_strength = 0.2
+
+[Preset]
+category = Area Specific Bloom Tint
+name = 30%
+condition = $bloom_enabled
+$bloom_tint_strength = 0.3
+
+[Preset]
+category = Area Specific Bloom Tint
+name = 40%
+condition = $bloom_enabled
+$bloom_tint_strength = 0.4
+
+[Preset]
+category = Area Specific Bloom Tint
+name = 50%
+condition = $bloom_enabled
+$bloom_tint_strength = 0.5
+
+[Preset]
+category = Area Specific Bloom Tint
+name = 60%
+condition = $bloom_enabled
+$bloom_tint_strength = 0.6
+
+[Preset]
+category = Area Specific Bloom Tint
+name = 70%
+condition = $bloom_enabled
+$bloom_tint_strength = 0.7
+
+[Preset]
+category = Area Specific Bloom Tint
+name = 80% (Recommended)
+condition = $bloom_enabled
+$bloom_tint_strength = 0.8
+
+[Preset]
+category = Area Specific Bloom Tint
+name = 90%
+condition = $bloom_enabled
+$bloom_tint_strength = 0.9
+
+[Preset]
+category = Area Specific Bloom Tint
+name = 100% (Native)
+condition = $bloom_enabled
+default = 1
+$bloom_tint_strength = 1.0
+
+
+# Shadow World Color Dampening (49865bd2e62efda1)
+# the alpha of the bloom color tint in the shadow world is used to darken the image
+# this is separated, because it is not affected by bloom strength at all
+
+[Preset]
+category = Shadow World Color Dampening
```

Comment:
Hard to explain without seeing it 😅
This shader dulls the colors (only in the shadow world) by shifting everything towards the red channel, effectivly desaturating the colors. This effect at 100% desaturates like default, at 0% it doesn't desaturate at all. I wanted to give more control over the effect with more options, because it can have a dramatic effect.
Also when you are on OpenGL you can see the result live. This itself took me some time to figure out, but with that you can just try and see what it does, no?

Reply:
Right, but I think it makes more sense to describe how heavy it is using textual options instead of percentage options. Maybe you can think of more text options and find some good presets? Its pretty likely that 60% of them are unusable.

From my experience, most people aren't too keen on using OpenGL (bad performance on AMD and Intel GPUs), so I'm more of a fan of having some steering and not providing too many options since it generally is overwhelming for people that are setting up a game for the first time.

Reply:
You should fix live shader update for Vulkan then :P

For the name, maybe Saturation, as this option changes the Color Saturation in shadow world and the other changes the Brightness in the shadow world. Though, actually it _is_ the other way around: The game desaturates and debrightenes the color already, the setting just reverts that. I tried to reflect that like that, but you're right, maybe it's easier to understand reversed.
It would be nice if you could try the pack live and see if it still doesn't make sense then.

I disagree on the text options. Changed a lot of rules myself to get what I want (which not everyone might be capable to do), why limit it to some arbitrary presets. This is a pure setting, controlling only one aspect. I would love to just have slider for it instead of those dropdown anyways. Instead I would rather give more explaination in the description for the settings.

I get that if you don't want to tinker, a preset can help, that's why I added the recommendation comments to the ones I ended up using, as an easy starting point.

Maybe I can combine simpler presets with more custom control with conditions. I have an idea for that, but changing that on mobile isn't that convinient currently. Can try that tomorrow.